woocommerce/google-listings-and-ads
# Adjust the budget recommendation conversion rate

### Changes proposed in this Pull Request:

Closes #1478 

This PR adjusts the budget recommendation conversion rate from 3.5% to 1.9%. The `wp_gla_budget_recommendations` table is populated using the CSV file: [budget-recommendations.csv](https://github.com/woocommerce/google-listings-and-ads/blob/718b65623981be0bb3d4d6bc25044ded3004d48b/data/budget-recommendations.csv). This PR updates the CSV with the new recommendations and it is adding a new migration class to adjust the budget recommendations for existing users.

- Project Thread – Adjust the budget recommendation conversion rate - pcTzPl-zZ-p2
- The new budget has been calculated using the spreadsheet provided in the project thead and updating the cell P4 (Website conversion rate) from 3.5% to 1.9%.
- [Old budget recommendations](https://github.com/woocommerce/google-listings-and-ads/blob/2db9cdc21e093963962c24e7bb0fb848a64b248f/data/budget-recommendations.csv) 
- [New budget recommendations](https://github.com/woocommerce/google-listings-and-ads/blob/718b65623981be0bb3d4d6bc25044ded3004d48b/data/budget-recommendations.csv)

### Detailed test instructions:


1. Check the table `wp_gla_budget_recommendations`, you should see the [old budget recommendations](https://github.com/woocommerce/google-listings-and-ads/blob/2db9cdc21e093963962c24e7bb0fb848a64b248f/data/budget-recommendations.csv).

2. Update the applicable version, the applicable version needs to be bigger than you current GLA version, for example 1.13.3

https://github.com/woocommerce/google-listings-and-ads/blob/c3b11dd6295c10c5c1fb22f3b3b3b03c2724c521/src/DB/Migration/Migration20220524T1653383133.php#L42

3. Update WC_GLA_VERSION,  if you used 1.13.3 in the step 2, the GLA_VERSION needs to be >= 1.13.3:

https://github.com/woocommerce/google-listings-and-ads/blob/cc008ad541e722f8c5108491135a9966d80c367e/google-listings-and-ads.php#L31


4. Refresh the dashboard page,  you should see the [new budget recommendations](https://github.com/woocommerce/google-listings-and-ads/blob/718b65623981be0bb3d4d6bc25044ded3004d48b/data/budget-recommendations.csv) in `wp_gla_budget_recommendations` table.


### Additional details:
- The GLA version can be also be updated by changing the values of: `gla_file_version` and `gla_db_version` in `wp_options`
- I saw that after changing the conversation rate to 1.9%, some budget recommendations are still the same, that is because the formula is rounding to a multiple of 5 or does not reach the minimum threshold. For example for currency GBP and country ES: `GBP,ES,5,10`

### Changelog entry
>Update - budget recommendation conversion rate
